pyVmomi/VmomiSupport.py

```diff
@@ -460,6 +460,8 @@ def __getattr__(self, name):
    def __eq__(self, other):
       if other is None:
          return False
+      elif type(other) is str:
```

Comment:
str is byte in python2 and unicode in python3. Maybe we should check for both byte and unicode? How about checking for six.binary_type and six.text_type here?